In [92]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

### 1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed.

In [97]:
diet_study = pd.read_csv("dietstudy.csv")

In [98]:
diet_study.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

In [100]:
diet_study.shape

(16, 13)

H0: NULL Hypothesis - mean of triglyceride before(tg0) = mean of triglyceride after(tg4) and mean of Weight before (wgt0) = mean of weight after (wgt4) H1: Alternate Hypothesis - mean of triglyceride before(tg0) != mean of triglyceride after(tg4) OR mean of Weight before (wgt0) != mean of weight after (wgt4) Confidence interval is 99% Test - Paired Sample t-test on both triglyceride and weight

In [101]:
stats.ttest_rel(a=diet_study.tg0,b=diet_study.tg4)

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

With 99% confidence interval ,pvalue(0.24) is not less than 1-Significance level(0.01) ,so NULL is accepted for triglyceride

In [102]:
stats.ttest_rel(a=diet_study.wgt0,b=diet_study.wgt4)

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)

With 99% confidence interval ,pvalue is less than 1-Significance level(0.01), so NULL is rejected

In [103]:
diet_study.wgt0.mean()

198.375

In [104]:
diet_study.wgt4.mean()

190.3125

Mean of weight shows that weight has decreased after new diet

#### RESULT: With 99% confidence we reject the hypothesis that mean triglyceride level before is not equal to after and mean weight before is equal to mean weight after the new diet.¶

### 2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales?

In [108]:
creditpromo=pd.read_csv("creditpromo.csv")

In [109]:
creditpromo.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [110]:
creditpromo.shape

(500, 3)

H0: NULL Hypothesis : mean sales of standard promotion = mean sales of New promotion H1: Alternate Hypothesis : mean sales of standard promotion != mean sales of New promotion Confidence interval is 95% Test : 2 Sample independent t-test

In [111]:
creditpromo_std=creditpromo[creditpromo['insert']=='Standard'].dollars

In [112]:
creditpromo_new=creditpromo[creditpromo['insert']=='New Promotion'].dollars

In [113]:
stats.ttest_ind(a=creditpromo_std,b=creditpromo_new,equal_var=False)

Ttest_indResult(statistic=-2.260422726464996, pvalue=0.024226348191648994)

With 95% confidence interval,pvalue(0.024) is less than 1-significance level(0.05) so we reject the null

In [114]:
creditpromo_std.mean()

1566.3890309659348

In [115]:
creditpromo_new.mean()

1637.4999830647992

Mean values shows that sales have increased after New promotion

#### RESULT: With 95% Confidence interval Mean sales before Promotion is not equal to Mean sales after Promotion and according that data Mean sales have increased after new sales Promotion.

### 3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). 

#### a. Is the overall population of Seed yield/plant (g) equals to 200?

In [17]:
pollination=pd.read_csv("pollination.csv")

In [19]:
pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

In [22]:
pollination.shape

(20, 4)

H0: NULL Hypothesis : Seed yield/plant = 200 H1: Alternate Hypothesis : Seed yield/plant not equal to 200 Confidence interval is 95% Test : 1 Sample independent t-test

In [23]:
stats.ttest_1samp(a=pollination.Seed_Yield_Plant,popmean=200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

With 95% confidence interval we see that pvalue(0.032) < 1-significance level(0.05) so WE REJECT THE NULL and see the stats

In [25]:
pollination.Seed_Yield_Plant.mean()

180.8035

Mean value shows that Seed yeild/plant is less than 200

##### RESULT: With 95% Confidence interval seed yeild/plant is not equal to 200 and accoding to data seed yeild/plant is less than 200

#### b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

In [26]:
pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

H0: Mean of Fruit Wt(by Hand Pollination) = Mean of Fruit Wt(By Natural Pollination) and Mean of Seed_Yield_Plant(by Hand Pollination) = Mean of Seed_Yield_Plant(By Natural Pollination) and Mean of Seedling Length(by Hand Pollination) = Mean of Seedling Length(By Natural Pollination) H1: Mean of Fruit Wt(by Hand Pollination) != Mean of Fruit Wt(By Natural Pollination) and Mean of Seed_Yield_Plant(by Hand Pollination) != Mean of Seed_Yield_Plant(By Natural Pollination) and Mean of Seedling Length(by Hand Pollination)!= Mean of Seedling Length(By Natural Pollination) Confidence interval is 95% Test : 2 Sample paired t-test on multiple data set

In [27]:
nat_fruitwt=pollination[pollination.Group=='Natural'].Fruit_Wt

In [28]:
hand_fruitwt=pollination[pollination.Group=='Hand'].Fruit_Wt

In [29]:
stats.ttest_rel(a=nat_fruitwt,b=hand_fruitwt)

Ttest_relResult(statistic=-16.003484109505287, pvalue=6.420421224715851e-08)

With 99% confidence interval,pvalue is less than 1-significance level(1-0.99=0.01) so we reject the null

In [30]:
pollination[pollination.Group=='Natural'].Fruit_Wt.mean()

1.848

In [31]:
pollination[pollination.Group=='Hand'].Fruit_Wt.mean()

2.5660000000000003

According to stats mean of Fruit Wt. by Hand(2.56) is greater than mean of fruit wt. by natural pollination(1.84)

In [32]:
nat_syp=pollination[pollination.Group=='Natural'].Seed_Yield_Plant

In [33]:
hand_syp=pollination[pollination.Group=='Hand'].Seed_Yield_Plant

In [34]:
stats.ttest_rel(a=nat_syp,b=hand_syp)

Ttest_relResult(statistic=-16.80079294985747, pvalue=4.198813240596638e-08)

In [35]:
pollination[pollination.Group=='Natural'].Seed_Yield_Plant.mean()

146.009

In [36]:
pollination[pollination.Group=='Hand'].Seed_Yield_Plant.mean()

215.598

Mean of Seed_yeild_plant by Hand(215.5) is greater than mean of Seed_yeild_plant by natural pollination(146)

In [37]:
nat_sl=pollination[pollination.Group=='Natural'].Seedling_length

In [38]:
hand_sl=pollination[pollination.Group=='Hand'].Seedling_length

In [39]:
stats.ttest_rel(a=nat_sl,b=hand_sl)

Ttest_relResult(statistic=-2.9765905136151303, pvalue=0.015535315669362245)

In [40]:
pollination[pollination.Group=='Natural'].Seedling_length.mean()

17.706999999999994

In [41]:
pollination[pollination.Group=='Hand'].Seedling_length.mean()

18.589999999999996

According to stats mean of Seedling_length by Hand(17.70) is greater than mean of Seedling_length by natural pollination(18.58)

##### RESULT: With 99% Confidence interval Natural Pollination and Hand Pollination under open field conditions are not equally effective and according to statistics Hand pollination is greater than Natural pollination

### 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently?

In [42]:
dvd=pd.read_csv("dvdplayer.csv")

In [43]:
dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [44]:
dvd.shape

(68, 2)

H0: mean dvd_un25 = dvd_25_34 = dvd_35_44 = dvd_45_54 = dvd_55_64 = dvd_65_over H1: atleast one of the mean is not equal to other Confidence interval is 99% Test : ANOVA


In [45]:
dvd_un25=dvd.dvdscore[dvd.agegroup=='Under 25']

In [46]:
dvd_25_34=dvd.dvdscore[dvd.agegroup=='25-34']

In [47]:
dvd_35_44=dvd.dvdscore[dvd.agegroup=='35-44']

In [48]:
dvd_45_54=dvd.dvdscore[dvd.agegroup=='45-54']

In [49]:
dvd_55_64=dvd.dvdscore[dvd.agegroup=='55-64']

In [50]:
dvd_65_over=dvd.dvdscore[dvd.agegroup=='65 and over']

In [51]:
stats.f_oneway(dvd_un25,dvd_25_34,dvd_35_44,dvd_45_54,dvd_55_64,dvd_65_over)

F_onewayResult(statistic=6.992526962676518, pvalue=3.087324905679639e-05)

In [52]:
dvd_un25.mean()

28.749228425584857

In [53]:
dvd_25_34.mean()

31.67800994228487

In [54]:
dvd_35_44.mean()

37.018058228022774

In [56]:
dvd_45_54.mean()

39.1183687318563

In [57]:
dvd_55_64.mean()

28.447335692086785

In [58]:
dvd_65_over.mean()

28.00279130552502

RESULT: With 99% Confidence interval we can say that consumers of various ages rated the design differently.

### 5. A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).

#### a. Is there any relationship in between labour force status with marital status?

#### b. Do you think educational qualification is somehow controlling the marital status?

#### c. Is happiness is driven by earnings or marital status?

In [116]:
Sample_survey=pd.read_csv("sample_survey.csv")
Sample_survey.head()

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   

   speduc          degree  ...        agecat childcat news1 news2  news3  \
0     NaN     High school  ...      55 to 64      1-2    No    No     No   
1     NaN  Junior college  ...      25 to 34     None    No    No    Yes   
2    16.0     High school  ...      35 to 44      1-2    No    No     No   
3     NaN     High school  ...  Less than 25     None    No    No     No   
4     NaN        Bachelor  ...      35 to 44     None    No    No     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  

[5 rows x 46 columns]

In [117]:
Sample_survey.isnull().sum()

id             0
wrkstat        1
marital        1
childs         7
age            4
educ          12
paeduc       791
maeduc       433
speduc      1521
degree        10
sex            0
race           0
born          13
parborn       20
granborn     202
income       329
rincome      983
polviews     141
cappun       233
postlife     766
happy         26
hapmar      1495
owngun       963
news         962
tvhours      495
howpaid     1485
ethnic       594
eth1         275
eth2        1730
eth3        2388
confinan     968
conbus      1011
coneduc      951
conpress     970
conmedic     957
contv        956
agecat         4
childcat       7
news1          0
news2          0
news3          0
news4          0
news5          0
car1           0
car2         566
car3        1160
dtype: int64

In [118]:
sample = Sample_survey.drop(columns=['howpaid','rincome','educ','agecat','childcat','childs','paeduc','maeduc','speduc','id','hapmar','owngun','tvhours','contv','news1', 'news2', 'news3', 'news4','news5', 'car1', 'car2', 'car3','ethnic', 'eth1','news','eth2', 'eth3', 'confinan', 'conbus', 'coneduc', 'conpress', 'conmedic','race', 'born', 'parborn', 'granborn','polviews', 'cappun', 'postlife'],axis=1)
sample.head()

wrkstat        marital   age          degree     sex  \
0  Working full time       Divorced  60.0     High school    Male   
1  Working part-time  Never married  27.0  Junior college  Female   
2  Working full time        Married  36.0     High school    Male   
3  Working full time  Never married  21.0     High school    Male   
4  Working full time  Never married  35.0        Bachelor  Female   

           income         happy  
0  $25000 or more  Pretty happy  
1  $15000 - 19999  Pretty happy  
2  $25000 or more    Very happy  
3  $15000 - 19999    Very happy  
4  $25000 or more  Pretty happy

In [119]:
sample.isnull().sum()

wrkstat      1
marital      1
age          4
degree      10
sex          0
income     329
happy       26
dtype: int64

In [120]:
sample.dtypes

wrkstat     object
marital     object
age        float64
degree      object
sex         object
income      object
happy       object
dtype: object

In [121]:
sample.wrkstat.value_counts()

Working full time          1568
Retired                     394
Keeping house               328
Working part-time           304
School                       79
Unemployed, laid off         58
Other                        54
Temporarily not working      46
Name: wrkstat, dtype: int64

In [122]:
sample.wrkstat.fillna(value = 'Working full time', inplace = True)

In [123]:
sample.marital.value_counts()

Married          1346
Never married     663
Divorced          446
Widowed           283
Separated          93
Name: marital, dtype: int64

In [124]:
sample.marital.fillna(value = 'Married', inplace = True)

In [125]:
sample.age.value_counts()

34.0    83
35.0    79
28.0    77
33.0    74
51.0    72
37.0    71
36.0    71
39.0    71
40.0    70
44.0    67
32.0    66
42.0    63
38.0    63
41.0    63
43.0    61
26.0    61
29.0    57
31.0    56
45.0    54
30.0    53
46.0    52
48.0    51
25.0    50
49.0    50
27.0    50
47.0    49
21.0    48
55.0    45
50.0    42
23.0    42
        ..
60.0    31
70.0    30
67.0    29
64.0    28
75.0    27
65.0    27
61.0    26
72.0    26
59.0    26
69.0    24
68.0    24
71.0    24
66.0    23
63.0    22
77.0    21
74.0    20
76.0    20
79.0    18
80.0    15
83.0    14
85.0    13
82.0    13
89.0    12
81.0    12
78.0    12
87.0    12
86.0     9
84.0     7
18.0     6
88.0     6
Name: age, Length: 72, dtype: int64

In [126]:
sample.age.fillna(value = 34.0, inplace = True)

In [127]:
sample.degree.value_counts()

High school       1500
Bachelor           478
LT High school     430
Junior college     209
Graduate           205
Name: degree, dtype: int64

In [128]:
sample.degree.fillna(value = 'High school', inplace = True)

In [129]:
sample.income.value_counts()

$25000 or more    1587
$20000 - 24999     247
$10000 - 14999     192
$15000 - 19999     179
$8000 TO 9999       59
$7000 TO 7999       47
LT $1000            36
$5000 TO 5999       35
$6000 TO 6999       33
$1000 TO 2999       32
$4000 TO 4999       32
$3000 TO 3999       24
Name: income, dtype: int64

In [130]:
sample.income.fillna(value = '$25000 or more', inplace = True)

In [131]:
sample.happy.value_counts()

Pretty happy     1575
Very happy        891
Not too happy     340
Name: happy, dtype: int64

In [132]:
sample.happy.fillna(value = 'Pretty happy', inplace = True)

In [133]:
sample.head()

wrkstat        marital   age          degree     sex  \
0  Working full time       Divorced  60.0     High school    Male   
1  Working part-time  Never married  27.0  Junior college  Female   
2  Working full time        Married  36.0     High school    Male   
3  Working full time  Never married  21.0     High school    Male   
4  Working full time  Never married  35.0        Bachelor  Female   

           income         happy  
0  $25000 or more  Pretty happy  
1  $15000 - 19999  Pretty happy  
2  $25000 or more    Very happy  
3  $15000 - 19999    Very happy  
4  $25000 or more  Pretty happy

In [134]:
sample['income'] = sample['income'].str.replace('$', '')
sample.head()

wrkstat        marital   age          degree     sex  \
0  Working full time       Divorced  60.0     High school    Male   
1  Working part-time  Never married  27.0  Junior college  Female   
2  Working full time        Married  36.0     High school    Male   
3  Working full time  Never married  21.0     High school    Male   
4  Working full time  Never married  35.0        Bachelor  Female   

          income         happy  
0  25000 or more  Pretty happy  
1  15000 - 19999  Pretty happy  
2  25000 or more    Very happy  
3  15000 - 19999    Very happy  
4  25000 or more  Pretty happy

(a)

H0 : Observed == Expected

In [135]:
wrkstat_marital_xtab = pd.crosstab(sample.wrkstat,sample.marital,margins = True)
wrkstat_marital_xtab

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      780            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1347            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1569  
Working part-time         304  
All                      2832

In [136]:
x2test = stats.chi2_contingency(observed= wrkstat_marital_xtab)
x2test

(729.7434446415614,
 1.435325149487234e-127,
 40,
 array([[5.16553672e+01, 1.56008475e+02, 7.67881356e+01, 1.07711864e+01,
         3.27768362e+01, 3.28000000e+02],
        [8.50423729e+00, 2.56843220e+01, 1.26419492e+01, 1.77330508e+00,
         5.39618644e+00, 5.40000000e+01],
        [6.20494350e+01, 1.87400424e+02, 9.22394068e+01, 1.29385593e+01,
         3.93721751e+01, 3.94000000e+02],
        [1.24413842e+01, 3.75752119e+01, 1.84947034e+01, 2.59427966e+00,
         7.89442090e+00, 7.90000000e+01],
        [7.24435028e+00, 2.18792373e+01, 1.07690678e+01, 1.51059322e+00,
         4.59675141e+00, 4.60000000e+01],
        [9.13418079e+00, 2.75868644e+01, 1.35783898e+01, 1.90466102e+00,
         5.79590395e+00, 5.80000000e+01],
        [2.47095339e+02, 7.46272246e+02, 3.67318856e+02, 5.15243644e+01,
         1.56789195e+02, 1.56900000e+03],
        [4.78757062e+01, 1.44593220e+02, 7.11694915e+01, 9.98305085e+00,
         3.03785311e+01, 3.04000000e+02],
        [4.46000000e+02, 1.347

In [137]:
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 729.7434446415614 and the p value is 1.435325149487234e-127


With 95% confidence interval,pvalue(1.435) is greater than 1-significance level(0.05) so we accept the null

##### RESULT: labour force status and marital status are relative

(b)

H0 : Observed == Expected

In [138]:
sample.head()

wrkstat        marital   age          degree     sex  \
0  Working full time       Divorced  60.0     High school    Male   
1  Working part-time  Never married  27.0  Junior college  Female   
2  Working full time        Married  36.0     High school    Male   
3  Working full time  Never married  21.0     High school    Male   
4  Working full time  Never married  35.0        Bachelor  Female   

          income         happy  
0  25000 or more  Pretty happy  
1  15000 - 19999  Pretty happy  
2  25000 or more    Very happy  
3  15000 - 19999    Very happy  
4  25000 or more  Pretty happy

In [139]:
degree_marital_xtab = pd.crosstab(sample.degree,sample.marital,margins = True)
degree_marital_xtab

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          244      690            370         58      148  1510
Junior college        45      109             46          3        6   209
LT High school        70      174             77         17       92   430
All                  446     1347            663         93      283  2832

In [140]:
x2test = stats.chi2_contingency(observed= degree_marital_xtab)
x2test

(123.18796051189827,
 6.0450187703046e-15,
 25,
 array([[  75.27824859,  227.35381356,  111.90466102,   15.6970339 ,
           47.76624294,  478.        ],
        [  32.28460452,   97.50529661,   47.99258475,    6.73199153,
           20.4855226 ,  205.        ],
        [ 237.80367232,  718.20974576,  353.50635593,   49.58686441,
          150.89336158, 1510.        ],
        [  32.91454802,   99.40783898,   48.92902542,    6.86334746,
           20.88524011,  209.        ],
        [  67.71892655,  204.52330508,  100.66737288,   14.12076271,
           42.96963277,  430.        ],
        [ 446.        , 1347.        ,  663.        ,   93.        ,
          283.        , 2832.        ]]))

In [141]:
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 123.18796051189827 and the p value is 6.0450187703046e-15


With 95% confidence interval,pvalue(6.045) is greater than 1-significance level(0.05) so we accept the null

##### Result: Yes,Educational Qualification is highly impacting the Marital Status.

(c)

In [142]:
happiness_income_xtab = pd.crosstab(sample.happy,sample.income,margins = True)
happiness_income_xtab

income         1000 TO 2999  10000 - 14999  15000 - 19999  20000 - 24999  \
happy                                                                      
Not too happy             7             39             33             40   
Pretty happy             20            109            120            157   
Very happy                5             44             26             50   
All                      32            192            179            247   

income         25000 or more  3000 TO 3999  4000 TO 4999  5000 TO 5999  \
happy                                                                    
Not too happy            151             9             9             6   
Pretty happy            1074            11            13            18   
Very happy               691             4            10            11   
All                     1916            24            32            35   

income         6000 TO 6999  7000 TO 7999  8000 TO 9999  LT 1000   All  
happy                                                                   
Not too happy            14            12             9       11   340  
Pretty happy             13            21            31       14  1601  
Very happy                6            14            19       11   891  
All                      33            47            59       36  2832

In [143]:
happiness_marital_xtab = pd.crosstab(sample.happy,sample.marital,margins = True)
happiness_marital_xtab

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        281      694            435         50      141  1601
Very happy           93      582            120         13       83   891
All                 446     1347            663         93      283  2832

In [144]:
x2test = stats.chi2_contingency(observed=happiness_income_xtab)
x2test

(177.41659890776808,
 1.3483634070060814e-20,
 36,
 array([[   3.84180791,   23.05084746,   21.49011299,   29.6539548 ,
          230.02824859,    2.88135593,    3.84180791,    4.2019774 ,
            3.96186441,    5.64265537,    7.08333333,    4.3220339 ,
          340.        ],
        [  18.09039548,  108.54237288,  101.19314972,  139.63524011,
         1083.16242938,   13.56779661,   18.09039548,   19.78637006,
           18.65572034,   26.57026836,   33.35416667,   20.35169492,
         1601.        ],
        [  10.06779661,   60.40677966,   56.31673729,   77.71080508,
          602.80932203,    7.55084746,   10.06779661,   11.01165254,
           10.38241525,   14.78707627,   18.5625    ,   11.32627119,
          891.        ],
        [  32.        ,  192.        ,  179.        ,  247.        ,
         1916.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   59.        ,   36.        ,
         2832.        ]]))

In [145]:
x3test = stats.chi2_contingency(observed= happiness_marital_xtab)
x3test

(261.0091307378115,
 6.6685158886813e-47,
 15,
 array([[  53.54519774,  161.71610169,   79.59745763,   11.16525424,
           33.9759887 ,  340.        ],
        [ 252.13488701,  761.49258475,  374.81038136,   52.57521186,
          159.98693503, 1601.        ],
        [ 140.31991525,  423.79131356,  208.59216102,   29.2595339 ,
           89.03707627,  891.        ],
        [ 446.        , 1347.        ,  663.        ,   93.        ,
          283.        , 2832.        ]]))

In [146]:
print("The chi square stat for happiness_income_xtab is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat for happiness_income_xtab is 177.41659890776808 and the p value is 1.3483634070060814e-20


In [147]:
print("The chi square stat for happiness_marital_xtab {} and the p value is {}".format(x3test[0],x3test[1]))

The chi square stat for happiness_marital_xtab 261.0091307378115 and the p value is 6.6685158886813e-47


##### Result: Yes happiness is driven by both Income and Marital Status,although Marital status is highly influencing the Happiness